wgpu-core/src/device/resource.rs

```diff
@@ -1086,6 +1086,20 @@ impl Device {
                         .saturating_sub(desc.range.base_array_layer),
                 });
 
+        let resolved_usage = {
+            let usage = desc.usage.unwrap_or(wgt::TextureUsages::empty());
+            if usage.is_empty() {
+                texture.desc.usage
+            } else if texture.desc.usage.contains(usage) {
+                usage
+            } else {
+                return Err(resource::CreateTextureViewError::InvalidTextureViewUsage {
+                    view: usage,
+                    texture: texture.desc.usage,
+                });
+            }
+        };
```

Comment:
We don't seem to be checking that the format supports the usage:

![image](https://github.com/user-attachments/assets/a3bb5142-ce1c-410b-93fb-e2806a9ca7df)

Reply:
There is some checking for render attachment later, but there is a lot more stuff missing/wrong ...

Anyway done in https://github.com/gfx-rs/wgpu/pull/6755/commits/c748f90a0944d8fc901b5764320045ce5f9375cb